components/api_version/src/lib.rs

```diff
@@ -311,10 +317,37 @@ mod tests {
                 Some(case.1),
                 case.2,
                 ApiVersion::V1ttl,
+                case.4
             );
         }
         for case in &cases {
-            assert_raw_value_encode_decode_identity(case.0, Some(case.1), case.3, ApiVersion::V2);
+            assert_raw_value_encode_decode_identity(case.0, Some(case.1), case.3, ApiVersion::V2, case.4);
+        }
+
+        // (user_value, expire_ts, ecoded_bytes_v2, api_version, is_delete)
+        let positive_cases = vec![
```

Comment:
En... api_version & is_delete fields are almost the same.

How about like this:
1. fn test_no_meta: test V1, V1TTL and V2 with cases without meta
2. fn test_ttl: test V1TTL and V2 with cases with ttl
3. fn test_meta: test V2 only, with cases has meta as combination of ttl & delete flag.